### PR TITLE
RXI-525 database unit tests

### PR DESCRIPTION
**Pull request by oleks-rip**

### CODE REVIEW

File: src/test/DB_test.cpp [hunk 1]
```diff
@@ -0,0 +1,555 @@
...
+#include <fmt/format.h>
+
+#include <filesystem>
+
+namespace xbwd {
+namespace tests {
+
+class DB_test : public beast::unit_test::suite
+{
+    // const char TEST_DB_NAME = "test.db";
```
Comment (seelabs) [inline src/test/DB_test.cpp:38]: I prefer to remove dead code like this.

File: src/test/DB_test.cpp [hunk 2]
```diff
@@ -0,0 +1,555 @@
...
+            return std::make_unique<DatabaseCon>(
+                ".",
+                db_init::xChainDBName(),
+                db_init::xChainDBPragma(),
+                db_init::xChainDBInit(),
+                j_);
+        }
+        catch (std::exception& ex)
+        {
+            // std::cerr << "Exception: " << ex.what() << std::endl;
```
Comment (seelabs) [inline src/test/DB_test.cpp:76]: Dead code.

**Review by seelabs — APPROVED** [review]
:+1: Your choice if you want to remove that dead code or not.

**Review by pwang200 — APPROVED** [review]
:+1: I am wondering if it is possible to have more negative tests, such as corrupted database.

File: src/test/DB_test.cpp (hunk 1 above)
Comment (oleks-rip) [reply src/test/DB_test.cpp]: fixed c872d3d

File: src/test/DB_test.cpp (hunk 2 above)
Comment (oleks-rip) [reply src/test/DB_test.cpp]: fixed c872d3d

**Comment by oleks-rip** [comment]
> 👍 I am wondering if it is possible to have more negative tests, such as corrupted database.
There is a test for bad DB - testBadDB. Don't know what we can do more. We relay on 3d party sqlite lib. If it open db - we consider it is good, if not - then we have no choice but consider it is corrupted.

**Pull request merged**